File: 11559.py
import sys

while True:
    nbhw_ = sys.stdin.readline()
    if nbhw_ == '':
        break;
    nbhw = nbhw_.split()

    s = (int)(nbhw[1])+1;
    for h in range(0,int(nbhw[2])):
        p = int(sys.stdin.readline())
        a_ = sys.stdin.readline()
        a = a_.split()

        for i in range(0,int(nbhw[3])):
            if (int(a[i])>=int(nbhw[0]) ):
                temp = p*int(nbhw[0]);
                if (temp<s):
                    s = temp
    if (s>int(nbhw[1]) ):
        print("stay home")
    else:
        print(s)

File: 11559.cpp
/*
 * =====================================================================================
 *
 *       Filename:  11559.cpp
 *
 *    Description:  
 *
 *        Version:  1.0
 *        Created:  08/04/2016 10:57:34
 *       Revision:  none
 *       Compiler:  gcc
 *
 *         Author:  Quang M. Hoang (QH), quanghm@gmail.com
 *        Company:  
 *
 * =====================================================================================
 */

#include <iostream>

using namespace std;

int main(){
    int n, // number of guests 
        b, // b
        h, // number of hotels
        w, // number of possible weeks
        p, // price per person
        a; // number of available beds 
    while (cin>>n){
        cin >> b >> h >> w;

        int sum = b+1;

        // read through the hotel list
        while (h--){

            // read the price per peson for that hotel
            cin >> p;
            int temp;

            // for each of the weekend
            for (int i=0;i<w;++i){
                cin >> a;
                // if the available beds >= number of persons
                if (a>=n){
                    temp = p*n;
                    if (temp<sum) sum = temp;
                }
            }
        }

            // check if sum is below b
            if (sum > b){
                cout<<"stay home\n";
            } else {
                cout<<sum<<"\n";
            }
    }
}
